ui/static/ui/js/listing.js

```diff
@@ -355,12 +279,14 @@ define('listing',
         });
       });
 
-      $("a#repo_page_status").click(function() {
-        var pageNum = $(this).data('page-num');
-        var maxPages = $(this).data('max-page-num');
-        navigateToPage(pageNum, maxPages, this);
-      });
+      var updatePage = function(newPageNum) {
+        var urlMap = URI(window.location).query(true);
+        urlMap.page = newPageNum;
+        window.location = "?" + URI().search(urlMap).query();
+      };
```

Comment:
Question: does this mean that we are tight to the necessary informations being in the URL?